src/libs/ui/progressitemdelegate.cpp

```diff
@@ -69,7 +72,28 @@ void ProgressItemDelegate::paint(QPainter *painter, const QStyleOptionViewItem &
     painter->translate(styleOption.rect.topRight());
     renderer->render(painter);
 
+    // Button
+    QScopedPointer<QPushButton> buttonRenderer(new QPushButton(tr("Cancel")));
+    buttonRenderer->resize(cancelButtonWidth, styleOption.rect.height());
+
+    painter->translate(progressBarWidth, 0);
+    buttonRenderer->render(painter);
     painter->restore();
 
     QStyledItemDelegate::paint(painter, styleOption, index);
 }
+
+bool ProgressItemDelegate::editorEvent(QEvent *event, QAbstractItemModel *model,
+                                       const QStyleOptionViewItem &option, const QModelIndex &index)
+{
+    QMouseEvent* mouseEvent = static_cast<QMouseEvent*>(event);
```

Comment:
```suggestion
    QMouseEvent *mouseEvent = static_cast<QMouseEvent *>(event);
```

Reply:
Now I understand the implication of this style. Thanks for pointing it out.

`int* a, b` is not int *a, *b`